packages/trader/src/Modules/Trading/Components/Form/TradeParams/Duration/duration-wrapper.jsx

```diff
@@ -5,63 +5,65 @@ import { connect } from 'Stores/connect';
 import { getDurationMinMaxValues } from 'Stores/Modules/Trading/Helpers/duration';
 import Duration from './duration.jsx';
 
-class DurationWrapper extends React.Component {
-    hasDurationUnit = (duration_unit, is_advanced_duration) => {
-        let duration_list = [...this.props.duration_units_list];
-
-        if (duration_list.length > 1 && !is_advanced_duration) {
+const DurationWrapper = props => {
+    const {
+        advanced_duration_unit,
+        advanced_expiry_type,
+        duration,
+        duration_unit,
+        expiry_type,
+        getDurationFromUnit,
+        is_advanced_duration,
+        onChange,
+        onChangeUiStore,
+        simple_duration_unit,
+        contract_expiry_type,
+        duration_min_max,
+        duration_units_list,
+        onChangeMultiple,
+    } = props;
+
+    const hasDurationUnit = (duration_type, is_advanced) => {
+        let duration_list = [...duration_units_list];
+
+        if (duration_list.length > 1 && !is_advanced) {
             duration_list = duration_list.filter(du => du.value === 'm' || du.value === 't');
         }
 
-        return duration_list.some(du => du.value === duration_unit);
+        return duration_list.some(du => du.value === duration_type);
     };
 
-    setDurationUnit() {
-        const new_duration_unit = this.props.duration_units_list[0].value;
-        const new_duration_value = this.props.getDurationFromUnit(new_duration_unit);
+    const setDurationUnit = () => {
+        const new_duration_unit = duration_units_list[0].value;
+        const new_duration_value = getDurationFromUnit(new_duration_unit);
 
-        this.props.onChangeUiStore({
-            name: `${this.props.is_advanced_duration ? 'advanced' : 'simple'}_duration_unit`,
+        onChangeUiStore({
+            name: `${is_advanced_duration ? 'advanced' : 'simple'}_duration_unit`,
             value: new_duration_unit,
         });
-        this.props.onChangeMultiple({
+        onChangeMultiple({
             duration_unit: new_duration_unit,
             duration: +new_duration_value,
         });
-    }
+    };
 
-    advancedHasWrongExpiry = () =>
-        this.props.is_advanced_duration && this.props.expiry_type !== this.props.advanced_expiry_type;
+    const advancedHasWrongExpiry = () => is_advanced_duration && expiry_type !== advanced_expiry_type;
```

Comment:
This can be applied as an effect with dependencies on `is_advanced_duration, expiry_type, and advanced_expiry_type`, to prevent running it on every render.

Reply:
I've changed line `109`, `advancedHasWrongExpiry` function runs only on init and dependencies changing.

Reply:
Do you mean it’s better to write a separated useEffect?